source/fundamentals/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1221 @@
+lowercase. This example stores the result as a new ``username`` field using
+the ``project()`` method.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   var lastName = current().getString("lastName");
+   var employeeID = current().getString("employeeID");
+
+   asList(project(fields(
+       computed("username", lastName
+           .append(employeeID)
+           .toLower())
+   )));
+   
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [ { $project: {
+       username: {
+           $toLower: { $concat: ["$lastName", "$employeeID"] } }
+   } } ]
+
+Type-Checking Operations
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can perform a type-check operation on a value of type ``MqlValue``
+using the methods described in this section.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Java Method
+     - Aggregation Pipeline Operator
+
+   * - `isArrayOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isArrayOr(com.mongodb.client.model.mql.MqlArray)>`__
+     - *No corresponding operator*
+
+   * - `isBooleanOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isBooleanOr(com.mongodb.client.model.mql.MqlBoolean)>`__
+     - *No corresponding operator*
+
+   * - `isDateOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isDateOr(com.mongodb.client.model.mql.MqlDate)>`__
+     - *No corresponding operator*
+
+   * - `isDocumentOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isDocumentOr(T)>`__
+     - *No corresponding operator*
+
+   * - `isIntegerOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isIntegerOr(com.mongodb.client.model.mql.MqlInteger)>`__
+     - *No corresponding operator*
+
+   * - `isMapOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isMapOr(com.mongodb.client.model.mql.MqlMap)>`__
+     - *No corresponding operator*
+
+   * - `isNumberOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isNumberOr(com.mongodb.client.model.mql.MqlNumber)>`__
+     - *No corresponding operator*
+
+   * - `isStringOr() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlValue.html#isStringOr(com.mongodb.client.model.mql.MqlString)>`__
+     - *No corresponding operator*
+
+Suppose you have a collection of rating data. An early version of the review
+schema allowed users to submit reviews without a star rating. You want to clean up
+the data, and convert any reviews without a star rating to be the minimum
+value of 1 star.
+
+The ``isNumberOr()`` method returns either the value of ``rating``, or
+a value of ``1`` if ``rating`` is not a number or is null. The
+``project()`` method returns this value as a new ``filteredRating`` field.
```

Comment:
It is odd to call this a filtered rating, since items are not being `filter`ed, but rather `map`ped to 1.